addon/helpers/optional.js

```diff
@@ -1,8 +1,8 @@
 import { helper } from 'ember-helper';
 
-export function optional([action]) {
+export function optional([action, ...args]) {
   if (typeof action === 'function') {
-    return action;
+    return action.bind(this, ...args);
```

Comment:
Certainly not `null` as that would set the context to `window`. I don't actually know what it should be. You should add a test using `this` in the action's context and then figure out what works.